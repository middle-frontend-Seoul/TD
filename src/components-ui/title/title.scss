.title {
  color: #ffffff;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  font-family: Roboto, Arial, Helvetica, sans-serif;

  &_size-small {
    font-size: 26px;
    line-height: 31px;
  }

  &_size-medium {
    font-size: $font-size-xxlg;
    line-height: 45px;
  }
}
