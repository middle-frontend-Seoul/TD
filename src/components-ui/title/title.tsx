import React, { FC, CSSProperties } from 'react';
import cn from 'classnames';

import './title.scss';

export type TitleSize = 'medium';

export interface ITitleProps {
  size?: TitleSize;
  style?: CSSProperties;
  className?: string;
  children: React.ReactNode;
}

export const Title: FC<ITitleProps> = ({
  className,
  children,
  style,
  size = 'medium',
}) => {
  const classes = cn(className, 'title', `title_size-${size}`);
  return (
    <h1 className={classes} style={style}>
      {children}
    </h1>
  );
};
