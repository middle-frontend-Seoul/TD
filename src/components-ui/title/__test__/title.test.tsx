import React from 'react';
import renderer from 'react-test-renderer';

import { Title } from '../title';

it('renders title', () => {
  const tree = renderer.create(<Title>hello test</Title>).toJSON();
  expect(tree).toMatchSnapshot();
});
