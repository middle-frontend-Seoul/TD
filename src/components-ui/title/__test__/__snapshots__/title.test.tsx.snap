// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<h1
  className="title title_size-medium"
>
  hello test
</h1>
`;
